### PR TITLE
Add modified_date support to the building block dataobject.

### PR DESCRIPTION
**Pull request by keithburgoyne**

### CODE REVIEW

**Comment by keithburgoyne** [comment]
SQL to run on all sites which use Building:
```
begin;
alter table block add modified_date timestamp not null default CURRENT_TIMESTAMP;
update block set modified_date = createdate;
commit;
```

**Comment by keithburgoyne** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;QdtBpLE5&#x2F;1475-1d-build-summary-support-into-chapter-api-endpoint"> 1d - Build summary support into chapter API endpoint</a>

**Comment by charleswaddell** [comment]
The SQL statement should probably be changed to:

`alter table block add modified_date timestamp not null default LOCALTIMESTAMP(0);`

The `CURRENT_TIMESTAMP` functions returns a timestamp + timezone. We don't store timezones with our timestamps so we should use the function that returns a timestamp without a timezone. In this case that function is `LOCALTIMESTAMP`.

**Comment by charleswaddell** [comment]
Has this SQL been run on all the live databases yet?

File: sql/tables/Block.sql [hunk 1]
```diff
@@ -2,6 +2,7 @@ create table Block (
 	id serial,
 
 	createdate timestamp not null,
+	modified_date timestamp not null,
```
Comment (charleswaddell) [inline sql/tables/Block.sql]: We should include the `default to LOCALTIMESTAMP(0)` here as well.

**Comment by gauthierm** [comment]
This still needs to be updated but is punted in favour of iOS 10 support release.

**Comment by keithburgoyne** [comment]
SQL above has been updated per instruction.

**Comment by charleswaddell** [comment]
> SQL above has been updated per instruction. It has not been run on any live sites. Is the usual process to do that once this PR is R+ed or merged?

For any changes that won't break the current live site we run them once the PR has been merged. That way we don't need to remember to do them when syncing the website live.

**Comment by keithburgoyne** [comment]
SQL has been run against dev and prod.

**Pull request merged**